applications/main/bad_usb/bad_usb_app.c

```diff
@@ -22,15 +25,45 @@ static void bad_usb_app_tick_event_callback(void* context) {
     scene_manager_handle_tick_event(app->scene_manager);
 }
 
+static void bad_usb_load_settings(BadUsbApp* app) {
+    File* settings_file = storage_file_alloc(furi_record_open(RECORD_STORAGE));
+    if(storage_file_open(settings_file, BAD_USB_SETTINGS_PATH, FSAM_READ, FSOM_OPEN_EXISTING)) {
+        char chr;
+        while((storage_file_read(settings_file, &chr, 1) == 1) &&
+              !storage_file_eof(settings_file) && !isspace(chr)) {
+            furi_string_push_back(app->keyboard_layout, chr);
+        }
+    }
+    storage_file_close(settings_file);
+    storage_file_free(settings_file);
+}
```

Comment:
Add layout file check here. If it is missing or has wrong size - clear `bad_usb->keyboard_layout` string.
Also it's better to clear config file if layout file path there is invalid